pkgs/applications/networking/cluster/kcp/default.nix

```diff
@@ -0,0 +1,72 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+# kubernetes-kcp,
+}:
+
+let
+  rev = "6a4e71a9d35c6906d3fdd46595e8b693c38669f5";
+in
+buildGoModule rec {
+  pname = "kubernetes-kcp";
+  version = "0.26.0";
+
+  src = fetchFromGitHub {
+    owner = "kcp-dev";
+    repo = "kcp";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-ZEgDeILo2weSAZgBsfR2EQyzym/I/+3P99b47E5Tfrw=";
+  };
+  vendorHash = "sha256-IONbTih48LKAiEPFNFdBkJDMI2sjHWxiqVbEJCskyio=";
+
+  subPackages = [ "cmd/kcp" ];
+
+  # TODO: The upstream has the additional version information pulled from go.mod
+  # dependencies.
+  ldflags = [
+    "-X k8s.io/client-go/pkg/version.gitCommit=${rev}"
+    "-X k8s.io/client-go/pkg/version.gitTreeState=clean"
+    "-X k8s.io/client-go/pkg/version.gitVersion=v${version}"
+    # "-X k8s.io/client-go/pkg/version.gitMajor=${KUBE_MAJOR_VERSION}"
+    # "-X k8s.io/client-go/pkg/version.gitMinor=${KUBE_MINOR_VERSION}"
+    "-X k8s.io/client-go/pkg/version.buildDate=unknown"
+    "-X k8s.io/component-base/version.gitCommit=${rev}"
```

Comment:
No need to, just mark unkown

Reply:
@Aleksanaa Could you elaborate why that's the case? Is this just customary, and/or because it's easier to update the reference for updates by simply using the tag?
This will be a slight diversion from the original binary, and when it is as simple as providing a commit reference, I thought it's worth keeping it around. (Also we should technically have the Kubernetes major / minor version reference, which would ensure the Nix package will be the same as the official binary.)

Reply:
Because setting rev specifically will make automatic update wrong, and not very useful as the version is sufficient to find the revision

Reply:
Thanks for the clarification! The automatic update support would be nice to have, and indeed the tag should give us the revision. But it is also possible that the tag gets removed and recreated for a different commit (though unlikely), and this revision reference acts as a safety net in a way I suppose.

In either case, it is also not a fast moving project AFAIAA, and it is something I can revisit later. The change has been made to remove the unnecessary let binding and git commit reference kept as unknown.